File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = {};
    const stack = [];
    let count = 0;
    
    for(let i=0; i<n; i+=1){
        if(!visited[i]){
            count+=1;
            stack.push(i);
        }
        
        while(true){
            if(stack.length===0){
                break;
            }
            const nextNode = stack.pop();
            visited[nextNode] = true;
            for(let j=0; j<n; j+=1){
                if(!visited[j]&&computers[nextNode][j]===1){
                    stack.push(j);
                    visited[j] = true;
                }
            }
        }
    }
    return count
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import defaultdict

def solution(n, computers):
    visited = set()
    tree = defaultdict(list)
    for i in range(n):
        tree[i] = []
        for j in range(n):
            if i!=j and computers[i][j]==1:
                tree[i].append(j)
    
    def dfs(node):
        visited.add(node)
        for i in tree[node]:
            if i not in visited:
                dfs(i)
                
    result = 0
    for t in tree:
        if t not in visited:
            dfs(t)
            result += 1
                
    return result